src/Resources/views/Pager/base_results.html.twig

```diff
@@ -25,6 +25,7 @@ file that was distributed with this source code.
     <select class="per-page small form-control" id="{{ admin.uniqid }}_per_page" style="width: auto">
         {% for per_page in admin.getperpageoptions %}
             <option {% if per_page == admin.datagrid.pager.maxperpage %}selected="selected"{% endif %} value="{{ admin.generateUrl(action, {'filter': admin.datagrid.values|merge({
+                (constant('Sonata\\AdminBundle\\Datagrid\\DatagridInterface::SORT_BY')): admin.datagrid.values[constant('Sonata\\AdminBundle\\Datagrid\\DatagridInterface::SORT_BY')].name,
```

Comment:
What if the value `admin.datagrid.values[constant('Sonata\\AdminBundle\\Datagrid\\DatagridInterface::SORT_BY')]` is not defined ?

Reply:
![image](https://user-images.githubusercontent.com/572270/121843039-ce8d3280-cce1-11eb-9d5e-1d25b149d440.png)

Reply:
The constant is alway defined, but not always the key in the array `admin.datagrid.values`

In the php code we're doing 
```
$values = $this->getValues();

        if (isset($values[DatagridInterface::SORT_BY]) && $values[DatagridInterface::SORT_BY] instanceof FieldDescriptionInterface) {
            $values[DatagridInterface::SORT_BY] = $values[DatagridInterface::SORT_BY]->getName();
        }
```

So we have to do a similar check in twig.

Reply:
OK, I said that mi fix was a quick and dirty, where you do this check? on generateUrl? I think something like this check may be on generateUrl and not on twig

Reply:
generateUrl is pretty straight forward, doing
```
return $this->getRouteGenerator()->generateUrl($this, $name, $parameters, $referenceType);
```

We already passing `'filter': admin.datagrid.values|merge(...)`, so it would be nice not having to transform the sortBy in the twig and instead correctly understand the value passed.

In the AbstractAdmin::buildDatagrid method there is the following code:
```
// transform DatagridInterface::SORT_BY from a string to a FieldDescriptionInterface for the datagrid.
        if (isset($filterParameters[DatagridInterface::SORT_BY]) && \is_string($filterParameters[DatagridInterface::SORT_BY])) {
            if ($this->hasListFieldDescription($filterParameters[DatagridInterface::SORT_BY])) {
                $filterParameters[DatagridInterface::SORT_BY] = $this->getListFieldDescription($filterParameters[DatagridInterface::SORT_BY]);
            } else {
                $filterParameters[DatagridInterface::SORT_BY] = $this->createFieldDescription(
                    $filterParameters[DatagridInterface::SORT_BY]
                );

                $this->getListBuilder()->buildField(null, $filterParameters[DatagridInterface::SORT_BY], $this);
            }
        }
```
Seems like we only handle the `\is_string($filterParameters[DatagridInterface::SORT_BY])` case. It would be nice to add the support of your case.

Reply:
Forget what I said, the value is not accessible in the request anymore, so we need to pass a string in the twig.

Maybe the best would be to use the `getPaginationParameters` method.
Changing 
```
admin.datagrid.values|merge({
                (constant('Sonata\\AdminBundle\\Datagrid\\DatagridInterface::PAGE')): 1,
                (constant('Sonata\\AdminBundle\\Datagrid\\DatagridInterface::PER_PAGE')): per_page})
            })
```
to 
```
admin.datagrid.getPaginationParameters(1)|merge({
                (constant('Sonata\\AdminBundle\\Datagrid\\DatagridInterface::PER_PAGE')): per_page})
            })
```